### PR TITLE
Generic k8s interface

### LINKED ISSUE #68: Implement generic k8s interface
Implement a set of generic methods for creating/deleting/getting/listing objects in k8s instead of the current set of methods per object type offered by the xk6-kubernetes extension. In this way the extension is simplified and the code base is reduced significantly as one set of methods cover all the resource types.

For example, for creating a job:
```
import { Kubernetes } from ‘xk6-kubernrtes’

const jobResource = {
    group: “batch”,
    version: “v1”,
    resource: “jobs” 
}

const jobDef = {
    apiVersion: "batch/v1",
    kind: "Job",
    metadata: {
        name: "demo-job",
    },
    spec: {
        template: {
            metadata: {
                labels: {
                    app: demo,
                }
            },
            spec: {
                containers: [
                    {
                        name:  "sleeper",
                        image: "busybox",
                        command: ["sh",  "-c", "sleep 3"]
                    }
                ]
            }
        }
   }
}

const k8s = new Kubernetes()
let job = k8s.create(jobResource, jobDef)
```
Similarly, for retrieving resources such a listing jobs:
```
import { Kubernetes } from ‘xk6-kubernetes’

const jobDef = {
    …                          // omitted for brevity
}

const k8s = new Kubernetes()
const namespace = ‘default’

let jobs = k8s.list(jobResource, namespace)
```

### PR DESCRIPTION
**Pull request by pablochacin**
Implement Generic interface for creating, retrieving, listing, deleting Kubernetes resources of any of a set of supported kinds such as Pod, Job, Secret, Service (among others)

Closes #68

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+type KubernetesConfig struct {
+	// Context for executing kubernetes operations
+	Context context.Context
+	// kubernetes rest config
+	Config *rest.Config
+	// Client is a pre-configured dynamic client. If provided, the rest config is not used
+	Client dynamic.Interface
+}
+
+// kubernetes Holds the reference to the helpers for interacting with kubernetes
```
Comment (dgzlopes) [inline pkg/api/api.go:60]:
```suggestion
// kubernetes holds the reference to the helpers for interacting with Kubernetes
```

File: README.md [hunk 2]
```diff
@@ -36,39 +36,128 @@ cd xk6-kubernetes
...
   console.log(`${pods.length} Pods found:`);
   pods.map(function(pod) {
-    console.log(`  ${pod.name}`)
+    console.log(`  ${pod.metadata.name}`)
   });
 }
 ```
 
-Using the `k6` binary with `xk6-kubernetes`, run the k6 test as usual:
+#### Creating a job using a yaml manifest
```
Comment (dgzlopes) [inline README.md:120]:
```suggestion
#### Creating a job using a YAML manifest
```

File: pkg/api/api.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	}
+
+	return &kubernetes{
+		ctx:        ctx,
+		client:     client,
+		serializer: yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme),
+	}, nil
+}
+
+// Apply creates a resource in a kubernetes cluster from a yaml manifest
```
Comment (dgzlopes) [inline pkg/api/api.go:90]:
```suggestion
// Apply creates a resource in a Kubernetes cluster from a YAML manifest
```

File: pkg/api/api.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+		Namespace(namespace).
+		Create(
+			k.ctx,
+			uObj,
+			metav1.CreateOptions{},
+		)
+	return err
+}
+
+// Create creates a resource in a kubernetes cluster from a yaml manifest
```
Comment (dgzlopes) [inline pkg/api/api.go:117]:
```suggestion
// Create creates a resource in a Kubernetes cluster from a YAML manifest
```

File: pkg/api/api.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+// maps kinds to api resources
+// TODO: complete with most common kinds
+func knownKinds(kind string) (schema.GroupVersionResource, error) {
+	kindMapping := map[string]schema.GroupVersionResource{
+		"ConfigMap":  {Group: "", Version: "v1", Resource: "configmaps"},
+		"Deployment": {Group: "apps", Version: "v1", Resource: "deployments"},
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply pkg/api/api.go:24]: Done. Also added PersitentVolume and PersitentVolumeClaim, as they are currently supported by the extension.

File: kubernetes_test.go [hunk 6]
```diff
@@ -456,3 +458,50 @@ if (k8s.services.list().length != initialCount) {
...
 }
+
+// TestGenericApiIsScriptable runs through creating, getting, listing and deleting an object
+func TestGenericApiIsScriptable(t *testing.T) {
+	t.Parallel()
+
+	rt := setupTestEnv(t)
+
+	_, err := rt.RunString(`
+const k8s = new Kubernetes()
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply kubernetes_test.go:469]: Added

File: pkg/api/api.go (hunk 4 above)
Comment (pablochacin) [reply pkg/api/api.go:117]: The comment is actually wrong as this method does not receive a YAML but an object with the resource specification

**Review by dgzlopes — APPROVED** [review]

File: examples/create_get_list_delete_pod.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    }
+}
+
+export default function(){
+        var created = k8s.create(podSpec)
+        console.log("pod '" + created.metadata.name +"' created")
+
+        var pod = k8s.get(podSpec.kind, podSpec.metadata.name, podSpec.metadata.namespace)
+        if (podSpec.metadata.name != pod.metadata.name) {
+                throw new Error("Fetch by name did not return the Service. Expected: " + podSpec.metadata.name + " but got: " + fetched.name)
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply examples/create_get_list_delete_pod.js]: Good catch. Fixed

**Pull request merged**